translation/translator.go

```diff
@@ -85,24 +87,42 @@ func (t *Translator) Get(key string, options ...translationcontract.Option) (str
 		return "", err
 	}
 
-	line := t.loaded[folder][locale][keyPart]
-	if line == "" {
-		fallbackFolder, fallbackLocale := parseKey(t.GetFallback())
-		// If the fallback locale is different from the current locale, we will
-		// load in the lines for the fallback locale and try to retrieve the
-		// translation for the given key.If it is translated, we will return it.
-		// Otherwise, we can finally return the key as that will be the final
-		// fallback.
-		if (folder+locale != fallbackFolder+fallbackLocale) && fallback {
-			var fallbackOptions translationcontract.Option
-			if len(options) > 0 {
-				fallbackOptions = options[0]
+	marshal, err := sonic.Marshal(t.loaded[folder][locale])
+	if err != nil {
+		return "", err
+	}
+	var keyParts []interface{}
+	for _, part := range strings.Split(keyPart, ".") {
+		keyParts = append(keyParts, part)
+	}
+
+	keyValue, err := sonic.Get(marshal, keyParts...)
+
+	if err != nil {
+		if err == ast.ErrNotExist {
+			fallbackLocale := t.GetFallback()
+			// If the fallback locale is different from the current locale, we will
+			// load in the lines for the fallback locale and try to retrieve the
+			// translation for the given key.If it is translated, we will return it.
+			// Otherwise, we can finally return the key as that will be the final
+			// fallback.
+			if (locale != fallbackLocale) && fallback {
+				var fallbackOptions translationcontract.Option
+				if len(options) > 0 {
+					fallbackOptions = options[0]
+				}
+				fallbackOptions.Fallback = translationcontract.Bool(false)
+				fallbackOptions.Locale = fallbackLocale
+				return t.Get(t.key, fallbackOptions)
 			}
-			fallbackOptions.Fallback = translationcontract.Bool(false)
-			fallbackOptions.Locale = fallbackLocale
-			return t.Get(fallbackFolder+"."+keyPart, fallbackOptions)
+			return t.key, nil
 		}
-		return t.key, nil
+		return "", err
+	}
+
+	line, err := keyValue.String()
```

Comment:
Q: Is it possible that `line` is empty?

Reply:
`line` empty means key doesn't exist which is handled just above of it.

Reply:
When `line` is empty, `Get` will return empty, I think it's unexpected.